extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1837,5 +1840,39 @@ describes.realWin(
         ).to.be.false;
       });
     });
+
+    describe('experiment for amp-story-load-first-page-only', () => {
+      let pages;
+      beforeEach(async () => {
+        toggleExperiment(win, 'amp-story-load-first-page-only', true);
+        pages = await createStoryWithPages(2, ['page-1', 'page-2'], false);
+        env.sandbox.stub(story, 'mutateElement').callsFake((fn) => fn());
+      });
+
+      it('should only load the active page with distance=0', async () => {
+        story.buildCallback();
+        await story.layoutCallback();
+
+        // Check page 0 is loaded with distance 0.
+        expect(pages[0].getAttribute('distance')).to.be.equal('0');
+      });
+
+      it('should load the inactive pages after the active page is loaded with distance=1', async () => {
+        const signals = new Signals();
+        pages[0].signals = () => signals;
+        story.buildCallback();
+        await story.layoutCallback();
+
+        // Check page 1 is not loaded.
+        expect(pages[0].hasAttribute('distance')).to.be.true;
+        expect(pages[1].hasAttribute('distance')).to.be.false;
+
+        signals.signal(CommonSignals.LOAD_END);
+        await nextTick();
+
+        // Check page 1 is loaded with distance 1.
+        expect(pages[1].getAttribute('distance')).to.be.equal('1');
+      });
+    });
```

Comment:
Re: can you add a test that was breaking before this PR, but passes now? Both your tests were passing before these changes, so you have no protection against a regression.

I think the correct test scenarios here are:

1. Active page loads on page load
2. Non active page doesn't load if active page hasn't loaded
3. Non active page loads after active page has loaded

Reply:
I was already checking those things, but I splitted the tests from 2 to 3 so they each cover a unique scenario from your list